org.mapstruct.eclipse/src/org/mapstruct/eclipse/internal/Bindings.java

```diff
@@ -39,25 +44,26 @@ private Bindings() {
      * @param type the type
      * @return the method names declared in the class or a super type of it
      */
-    public static Set<String> findAllMethodNames(ITypeBinding type) {
-        Set<String> result = new HashSet<String>();
+    public static Collection<String> findAllMethodNames(ITypeBinding type) {
```

Comment:
That was so long ago, I have noooooo idea... but I'm with you - the declared return type should be `Set`